packages/graphql/src/translate/queryAST/factory/OperationFactory.ts

```diff
@@ -229,76 +261,141 @@ export class OperationsFactory {
     }
 
     public createConnectionOperationAST(
-        relationship: RelationshipAdapter,
+        entityOrRel: ConcreteEntityAdapter | RelationshipAdapter,
         resolveTree: ResolveTree,
         context: Neo4jGraphQLTranslationContext
     ): ConnectionReadOperation | CompositeConnectionReadOperation {
-        const target = relationship.target;
+        const entity = entityOrRel instanceof RelationshipAdapter ? entityOrRel.target : entityOrRel;
+        const relationship = entityOrRel instanceof RelationshipAdapter ? entityOrRel : undefined;
         const directed = Boolean(resolveTree.args.directed) ?? true;
         const resolveTreeWhere: Record<string, any> = isObject(resolveTree.args.where) ? resolveTree.args.where : {};
 
-        if (isConcreteEntity(target)) {
-            checkEntityAuthentication({
-                entity: target.entity,
-                targetOperations: ["READ"],
-                context,
-            });
-
-            const operation = new ConnectionReadOperation({ relationship, directed, target });
+        if (relationship) {
```

Comment:
Could we extract this if and its nested if-checks into separate functions? This is really hard to follow..